src/containers/pages/IRS/performanceReport/plans/tests/index.test.tsx

```diff
@@ -0,0 +1,146 @@
+import { genericFetchPlans, GenericPlan } from '../../../../../../store/ducks/generic/plans';
+import * as fixtures from '../../../../../../store/ducks/generic/tests/fixtures';
+
+jest.mock('../../../../../../configs/env');
+/* tslint:disable-next-line no-var-requires */
+const fetch = require('jest-fetch-mock');
+
+const history = createBrowserHistory();
+
+describe('IRS performance report', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders without crashing', () => {
+    const props = {
+      history,
+      location: {
+        hash: '',
+        pathname: PERFORMANCE_REPORT_IRS_PLAN_URL,
+        search: '',
+        state: undefined,
+      },
+      match: {
+        isExact: true,
+        params: {},
+        path: `${PERFORMANCE_REPORT_IRS_PLAN_URL}/`,
+        url: `${PERFORMANCE_REPORT_IRS_PLAN_URL}/`,
+      },
+      plans: fixtures.plans as GenericPlan[],
+    };
+    shallow(
+      <Router history={history}>
+        <IRSPlanPerfomenceReport {...props} />
+      </Router>
+    );
+  });
+
+  it('renders plan list view correctly', async () => {
+    fetch.mockResponseOnce(JSON.stringify({}));
+    const props = {
+      history,
+      location: {
+        hash: '',
+        pathname: PERFORMANCE_REPORT_IRS_PLAN_URL,
+        search: '',
+        state: undefined,
+      },
+      match: {
+        isExact: true,
+        params: {},
+        path: `${PERFORMANCE_REPORT_IRS_PLAN_URL}/`,
+        url: `${PERFORMANCE_REPORT_IRS_PLAN_URL}/`,
+      },
+      plans: fixtures.plans as GenericPlan[],
+    };
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <IRSPlanPerfomenceReport {...props} />
+        </Router>
+      </Provider>
+    );
+    await act(async () => {
+      await new Promise(resolve => setImmediate(resolve));
+      wrapper.update();
+    });
+    expect(wrapper.find('h3.page-title').text()).toEqual('IRS Performance Reporting');
+    const tableHeader = wrapper.find('.thead .tr .th');
+    Array(tableHeader.length)
+      .fill('')
+      .forEach((_, i) => {
+        expect(tableHeader.at(i).text()).toMatchSnapshot(`table header-${i + 1}`);
+      });
+    expect(wrapper.find('GenericPlansList').length).toBe(1);
+    renderTable(wrapper, 'the full rendered rows');
+    // have searchbar
+    expect(wrapper.find('.search-input-wrapper').length).toEqual(1);
+    // have top and bottom pagination
+    expect(wrapper.find('.pagination').length).toEqual(2);
+    // have height and columns filters
+    expect(
+      wrapper
+        .find('.filter-bar-btns span')
+        .at(0)
+        .text()
+    ).toEqual('Row Height');
+    expect(
+      wrapper
+        .find('.filter-bar-btns span')
+        .at(1)
+        .text()
+    ).toEqual('Customize Columns');
+    wrapper.unmount();
+  });
+
+  it('renders plans correctly when connected to store', async () => {
+    store.dispatch(genericFetchPlans(fixtures.plans as GenericPlan[]));
```

Comment:
Nitpick: instead of dispatching here; what if we resolved the data in the service mock and have the  component dispatch the data as it would when built